packages/remix/test/integration/test/server/loader.test.ts

```diff
@@ -131,4 +131,22 @@ describe('Remix API Loaders', () => {
       },
     });
   });
+
+  it('makes sure scope does not bleed between requests', async () => {
```

Comment:
Did this test previously fail? IOW, would it actually detect scope bleed if we were to accidentally re-introduce it?

Reply:
This test did previously fail - but now it seems to be sometimes passing 🤔, worried it's a flaky test then. @lobsterkatie any ideas for a good test against scope bleed?

Reply:
Hmmm... I would assume we basically have to force them to be simultaneous, right? This test would pass regardless, as long as there were only one request/transaction at at time, so maybe the flakiness is coming from the requests just getting through so quickly that they end up being sequential. Could you introduce a variable delay into each request, and set them off at a set interval, to guarantee they'd all be in flight at the same time? I'm picturing something like (assuming 5 requests):

Request 1: Starts at timestamp 0, waits 5 seconds to set its tag, waits another second, finishes
Request 2: Starts at timestamp 1 second, waits 4 seconds to set its tag, waits another second, finishes
...
Request 5: Starts at timestamp 4 seconds, waits 1 second to set its tag, waits another second, finishes

That way, you know that 5 seconds in, all five requests should be trying to set their tags more or less at the same moment, and should also all be finishing at more or less the same time (meaning they'd all be trying to grab scope data to attach to the event roughly simultaneously). 

(You might also consider introducing a tiny bit of randomness (wait anywhere between 0.99 and 1.01 seconds to start each request, and the do it again for finishing, for example), just so that the order in which requests set and get their tags is mixed up.)

If everything still comes through cleanly, then I think we'd've proved the point, right?

Reply:
Yup, and we've tested on express as well. Thanks Katie, used a simple version of your logic here.

I tried the randomness, but it wasn't working that well, so elected to do this.